packages/dtslint/src/rules/npm-naming.ts

```diff
@@ -0,0 +1,234 @@
+    },
+    schema: [
+      {
+        oneOf: [
+          {
+            type: "object",
+            properties: {
+              mode: {
+                type: "string",
+                enum: [Mode.NameOnly],
+              },
+              "single-line": {
+                description: "Whether to print error messages in a single line. Used for testing.",
+                type: "boolean",
+              },
+            },
+            required: ["mode"],
+          },
+          {
+            type: "object",
+            properties: {
+              mode: {
+                type: "string",
+                enum: [Mode.Code],
+              },
+              errors: {
+                type: "array",
+                items: {
+                  type: "array",
+                  items: [
+                    {
+                      description: "Name of the check.",
+                      type: "string",
+                      enum: [ErrorKind.NeedsExportEquals, ErrorKind.NoDefaultExport] as ExportErrorKind[],
+                    },
+                    {
+                      description: "Whether the check is enabled or disabled.",
+                      type: "boolean",
+                    },
+                  ],
+                  minItems: 2,
+                  maxItems: 2,
+                },
+              },
+              "single-line": {
+                description: "Whether to print error messages in a single line. Used for testing.",
+                type: "boolean",
+              },
+            },
+            required: ["mode"],
+          },
+        ],
+      },
+    ],
+  },
+  // TODO: This code is a modified version of the old TSLint rule,
+  // and still primarily uses TypeScript nodes. It would be better
+  // to switch it to using TSESTree nodes like other ESLint rules.
+  create(context) {
+    const options = toCriticOptions(parseOptions(context.options));
+    const services = ESLintUtils.getParserServices(context, true);
+    const sourceCode = context.getSourceCode();
+    const sourceFile = services.esTreeNodeToTSNodeMap.get(sourceCode.ast);
+
+    const { text } = sourceFile;
+    const lookFor = (search: string, error: CriticError) => {
+      const idx = text.indexOf(search);
+      if (idx !== -1) {
+        reportRange(idx, search.length, error);
+      }
+    };
+
+    if (isMainFile(sourceFile.fileName, /*allowNested*/ false)) {
+      try {
+        const optionsWithSuggestions = toOptionsWithSuggestions(options);
+        const diagnostics = critic(sourceFile.fileName, /* sourcePath */ undefined, optionsWithSuggestions);
+        const errors = filterErrors(diagnostics);
+        for (const error of errors) {
+          switch (error.kind) {
+            case ErrorKind.NoMatchingNpmPackage:
+            case ErrorKind.NoMatchingNpmVersion:
+            case ErrorKind.NonNpmHasMatchingPackage:
+              lookFor("// Type definitions for", error);
```

Comment:
note to self: this will have to change in the monorepo branch since DT headers will have moved to package.json (although I think this is just intended to put the error at the beginning of the file).